Userland/Games/BrickGame/BrickGame.cpp

```diff
@@ -0,0 +1,511 @@
+        }
+        return false;
+    }
+
+    bool place_into(Well& well)
+    {
+        for (size_t i = 0; i < shape_size; ++i) {
+            well[m_pos.y() + i] |= block_row(i);
+        }
+        return true;
+    }
+};
+
+class Bricks final {
+    Well m_well;
+    Block m_block;
+    unsigned m_level {};
+    unsigned m_score {};
+    bool m_is_game_over {};
+    AK::Time m_last_update;
+
+    struct LevelMap final {
+        unsigned const m_score;
+        AK::Time const m_delay;
+    };
+
+    static constexpr AK::Array<LevelMap, 14> s_level_map = {
+        LevelMap { 0, AK::Time::from_milliseconds(36000 / 60) },
+        LevelMap { 1500, AK::Time::from_nanoseconds(32000 / 60) },
+        LevelMap { 8000, AK::Time::from_nanoseconds(29000 / 60) },
+        LevelMap { 20000, AK::Time::from_nanoseconds(25000 / 60) },
+        LevelMap { 40000, AK::Time::from_nanoseconds(22000 / 60) },
+        LevelMap { 75000, AK::Time::from_nanoseconds(18000 / 60) },
+        LevelMap { 100000, AK::Time::from_nanoseconds(15000 / 60) },
+        LevelMap { 200000, AK::Time::from_nanoseconds(11000 / 60) },
+        LevelMap { 300000, AK::Time::from_nanoseconds(7000 / 60) },
+        LevelMap { 400000, AK::Time::from_nanoseconds(5000 / 60) },
+        LevelMap { 500000, AK::Time::from_nanoseconds(4000 / 60) },
+        LevelMap { 600000, AK::Time::from_nanoseconds(3000 / 60) },
+        LevelMap { 700000, AK::Time::from_nanoseconds(2000 / 60) },
+        LevelMap { 10000000, AK::Time::from_nanoseconds(1000 / 60) }
+    };
+
+    void set_current_block(Block const& block)
+    {
+        if (!block.has_collision(m_well)) {
+            m_block = block;
+        }
+    }
+
+public:
+    Bricks() { reset(); }
+
+    [[nodiscard]] static constexpr size_t width() { return Well::num_columns(); }
+    [[nodiscard]] static constexpr size_t height() { return Well::num_rows(); }
+
+    [[nodiscard]] bool is_game_over() const { return m_is_game_over; }
+    [[nodiscard]] unsigned score() const { return m_score; }
+    [[nodiscard]] unsigned level() const { return m_level; }
+
+    [[nodiscard]] bool new_block()
+    {
+        m_block = Block().random_shape();
+        m_is_game_over = m_block.has_collision(m_well);
+        return !m_is_game_over;
+    }
+    [[nodiscard]] bool operator[](Position pos) const
+    {
+        return m_well[pos] || m_block[pos];
+    }
+
+    void rotate_left() { set_current_block(Block(m_block).rotate_left()); }
+    void rotate_right() { set_current_block(Block(m_block).rotate_right()); }
+
+    void move_left() { set_current_block(Block(m_block).move_left()); }
+    void move_right() { set_current_block(Block(m_block).move_right()); }
+    bool move_down()
+    {
+        auto const block = Block(m_block).move_down();
+        if (block.has_collision(m_well)) {
+            m_block.place_into(m_well);
+            return new_block();
+        }
+        m_block = block;
+        return true;
+    }
+    bool move_down_fast()
```

Comment:
Should be new-lines between any set of new blocks like this -- or at least be consistent.